dependent-usage-analyzer/index.js

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-console */
+const parser = require('@babel/parser');
+const fs = require('fs');
+const walk = require('babel-walk');
+const glob = require('glob');
+const { Command } = require('commander');
+const path = require('path');
+
+function getProjectFiles(dir) {
+  // Common project directories to ignore
+  const ignore = [
+    `${dir}/**/node_modules/**`,
+    `${dir}/dist/**`,
+    `${dir}/public/**`,
+    `${dir}/coverage/**`,
+    `${dir}/**/*.config.*`,
+  ];
+  // Gather all js and jsx source files
+  return glob.sync(`${dir}/**/*.{js,jsx}`, { ignore });
+}
+
+function getPackageInfo(dir) {
+  try {
+    // Package lock contains the actual Paragon version rather than a range in package.json
+    const { dependencies } = JSON.parse(fs.readFileSync(`${dir}/package-lock.json`, { encoding: 'utf-8' }));
+    const { name, repository } = JSON.parse(fs.readFileSync(`${dir}/package.json`, { encoding: 'utf-8' }));
+
+    return {
+      version: dependencies && dependencies['@edx/paragon'] ? dependencies['@edx/paragon'].version : 'unknown',
```

Comment:
probably for later, but I think we'll also want to look in `peerDependencies`.

Reply:
Right now the version we're pulling out is from package-lock so that we get the exact version used. Agreed I think it makes sense to fallback to peer dependency ranges there's no direct dependency in the package-lock.